src/docfx/build/xref/XrefResolver.cs

```diff
@@ -158,14 +158,20 @@ public XrefMapModel ToXrefMapModel()
                 .OrderBy(xref => xref.Uid)
                 .ToArray();
 
-            externalXrefs = _jsonSchemaTransformer().GetValidateExternalXrefs();
+            var monikerGroups = new Dictionary<string, MonikerList>(
+                from item in references
+                let monikerGroup = item.MonikerGroup
+                where !string.IsNullOrEmpty(monikerGroup)
+                orderby monikerGroup
+                group item by monikerGroup into g
+                select new KeyValuePair<string, MonikerList>(g.Key, g.First().Monikers));
```

Comment:
some minor thinking: can this be shared with publish model somehow?

Reply:
Thought about it, did not find a proper utility for it, though.

Reply:
What about caching a the moniker map<hash, monikers> here https://github.com/dotnet/docfx/blob/v3/src/docfx/build/moniker/MonikerProvider.cs#L68-L71?

Reply:
Actually, there is not much to share in this case.
We have different items, both item happen to have `MonikerGroup` and `Monikers`, and what we are doing is to construct a mapping from `MonikerGroup` to `Monikers`, `MonikerGroup` is computed already. No extra calculation or cache needed.